src/components/control/select.jsx

```diff
@@ -195,6 +196,30 @@ const Select = forwardRef(function Select(props, ref) {
       .forEach(_ref => _ref.current.setSelected(JSON.stringify(value) === JSON.stringify(_ref.current.value)));
   }, [value, showOptions]);
 
+  useEffect(() => {
+    if (hasBeenCleared) {
+      refs.input.current.focus();
+    }
+
+    setBeenCleared(false);
+  }, [hasBeenCleared]);
+
+  const currentValue = searchValue ?? defaultValue ?? '';
+  const shouldRenderSpecialChild = props.specialChildrenRender && defaultValue;
+
+  function renderSpecialChild() {
```

Comment:
What is the purpose of a special child? I'd like to understand the use-case. I'm concerned we're creating a one-off autocompleter.

Reply:
It allows you to render a child as a selected component.

Reply:
So it will determine based on keys what is selected and render that instead of the input. Like this:
<img width="584" alt="Screen Shot 2022-07-26 at 10 10 17 AM" src="https://user-images.githubusercontent.com/2433091/181056752-cd08a9ff-7833-4a70-96d1-69e6ac479cdd.png">

Reply:
It's fairly generic, you just need id as keys in your children and values to be of the shape `{ id: <id> ... }`.